### PR TITLE
Implement Zod schemas for Filters in utils [BOOST-4470]

### PR DESCRIPTION
**Pull request by sammccord**
- Implement Zod schemas for Filters in utils
- Fix several explicit any's to resolve ts issues
- Update Mint params schema to match number or FilterOperator
- Update zod to support readonly usage in abitype

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "noImplicitOverride": true,
     "noUnusedLocals": true,
     "noUnusedParameters": true,
+    "noImplicitAny": false,
```
Comment (sammccord) [inline tsconfig.json:15]: this was giving me guff trying to build, figured that at this point any any's we have right now are fine and we're unlikely to introduce a great deal more core functionality

File: packages/utils/src/types/filters.ts [hunk 2]
```diff
@@ -39,41 +22,175 @@ export type NumericOperator =
...
+  .object({
+    $abiParams: AbiSchema,
+  })
+  .catchall(z.lazy(() => FilterSchema))
+
+export type Filter =
+  | Primitive
+  | FilterObject
+  | FilterArray
+  | FilterOperator
```
Comment (sammccord) [inline packages/utils/src/types/filters.ts:176]: @Quazia is this correct? I added the `FilterOperator` to `Filter` union type, and schema. Unsure if it's the move.

Comment (Quazia) [reply packages/utils/src/types/filters.ts:176]: I think that is correct - some of the issues before may have been from _not_ having it in this.

**Review by Quazia — APPROVED** [review]
Looks good - this must have been a tough one 😅 

Just for my edification why'd you end up casting the context as any when passing it through in the handlers?

**Comment by sammccord** [comment]
> Looks good - this must have been a tough one 😅
> 
> Just for my edification why'd you end up casting the context as any when passing it through in the handlers?

The context type unions don't always overlap with the operator type unions, especially when it comes to `Record<string, unknown>` vs transaction interfaces.

**Pull request merged**